macros/edr/dbt_artifacts/upload_artifacts_to_table.sql

```diff
@@ -18,4 +18,60 @@
     {%- if should_commit -%}
         {% do adapter.commit() %}
     {%- endif -%}
-{% endmacro %}
\ No newline at end of file
+{% endmacro %}
+
+{% macro upload_csv_artifacts_to_table(table_relation, artifacts, flatten_artifact_callback) %}
+    {% set output_csv_path = elementary.get_target_path(table_relation.identifier ~ '.csv') %}
+    {% set flatten_artifact_dicts = [] %}
+    {% for artifact in artifacts %}
+        {% set flatten_artifact_dict = flatten_artifact_callback(artifact) %}
+        {% set flatten_artifact_rendered_dict = {} %}
+        {% for key in flatten_artifact_dict %}
+          {% set value = flatten_artifact_dict[key] %}
+          {% if value %}
```

Comment:
Don't really like this, but `flatten_test()` is currently used for both `elementary_test_results` and `dbt_tests`.
This is a problem since these tables don't share all the columns and therefore certain columns, such as `compiled_code`, will try to insert into `dbt_tests` in which it doesn't exist.